File: marvelHeroes/src/main/java/com/futureworkshops/marvelheroes/domain/rx/scheduler/WorkerSchedulerProvider.java
/*
 * Copyright (c) 2018 FutureWorkshops. All rights reserved.
 */

package com.futureworkshops.marvelheroes.domain.rx.scheduler;

import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class WorkerSchedulerProvider implements SchedulersProvider {

    @Override
    public Scheduler io() {
        return Schedulers.io();
    }

    @Override
    public Scheduler ui() {
        return AndroidSchedulers.mainThread();
    }
}

File: marvelHeroes/src/main/java/com/futureworkshops/marvelheroes/data/network/rx/scheduler/WorkerSchedulerProvider.kt
/*
 * Copyright (c) 2018 FutureWorkshops. All rights reserved.
 */

package com.futureworkshops.marvelheroes.data.network.rx.scheduler

import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

class WorkerSchedulerProvider : SchedulersProvider {
    
    override fun io(): Scheduler {
        return Schedulers.io()
    }
    
    override fun ui(): Scheduler {
        return AndroidSchedulers.mainThread()
    }
}
